src/Symfony/Component/HttpKernel/EventListener/LocaleListener.php

```diff
@@ -65,6 +69,10 @@ private function setLocale(Request $request)
         if ($locale = $request->attributes->get('_locale')) {
             $request->setLocale($locale);
         }
+
+        if ($this->useAcceptLanguageHeader && !$locale && $this->enabledLocales && ($preferredLanguage = $request->getPreferredLanguage($this->enabledLocales))) {
```

Comment:
Nitpicking (to simplify the code and remove a useless check).

```suggestion
        elseif ($this->useAcceptLanguageHeader && $this->enabledLocales && ($preferredLanguage = $request->getPreferredLanguage($this->enabledLocales))) {
```